src/librustc_typeck/error_codes.rs

```diff
@@ -2425,6 +2425,83 @@ struct Bar<S, T> { x: Foo<S, T> }
 ```
 "##,
 
+E0307: r##"
+This error indicates that the `self` parameter in a method has an invalid
+"reciever type".
+
+Methods take a special first parameter, of which there are three variants:
+`self`, `&self`, and `&mut self`. The type `Self` acts as an alias to the
```

Comment:
I think these should be *explained as* sugar for `self: Self`, `self: &Self`, and `self: &mut Self` respectively.